examples/explainable_ai/steps/trainers.py

```diff
@@ -0,0 +1,149 @@
+        """
+
+        super().__init__()
+
+        # Set our init args as class attributes
+        self.learning_rate = learning_rate
+
+        # Hardcode some dataset specific attributes
+        self.model = model
+        self.num_classes = num_labels
+
+        self.dataloader_train = dataloader_train
+        self.dataloader_test = dataloader_test
+
+        self.val_accuracy = Accuracy(task="multiclass", num_classes=self.num_classes)
+        self.test_accuracy = Accuracy(task="multiclass", num_classes=self.num_classes)
+
+    def forward(  # pylint: disable = (arguments-differ)
+        self, x: torch.Tensor
+    ) -> torch.Tensor:
+        x = self.model(x)
+        return F.log_softmax(x, dim=1)
+
+    def training_step(  # pylint: disable = (arguments-differ)
+        self,
+        batch: torch.Tensor,
+        batch_idx: int,  # pylint: disable = (unused-argument)
+    ) -> torch.Tensor:
+        samples, targets = batch
+        logits = self(samples)
+        loss = F.nll_loss(logits, targets)
+        return loss
+
+    def validation_step(  # pylint: disable = (arguments-differ)
+        self,
+        batch: torch.Tensor,
+        batch_idx: int,  # pylint: disable = (unused-argument)
+    ) -> None:
+        samples, targets = batch
+        logits = self(samples)
+        loss = F.nll_loss(logits, targets)
+        preds = torch.argmax(logits, dim=1)  # pylint: disable = (no-member)
+        self.val_accuracy.update(preds, targets)  # pylint: disable = (no-member)
+
+        # Calling self.log will surface up scalars for you in TensorBoard
+        self.log("val_loss", loss, prog_bar=True)
+        self.log("val_acc", self.val_accuracy, prog_bar=True)
+
+    def test_step(  # pylint: disable = (arguments-differ)
+        self,
+        batch: torch.Tensor,
+        batch_idx: int,  # pylint: disable = (unused-argument)
+    ) -> None:
+        samples, targets = batch
+        logits = self(samples)
+        loss = F.nll_loss(logits, targets)
+        preds = torch.argmax(logits, dim=1)  # pylint: disable = (no-member)
+        self.test_accuracy.update(preds, targets)  # pylint: disable = (no-member)
+
+        # Calling self.log will surface up scalars for you in TensorBoard
+        self.log("test_loss", loss, prog_bar=True)
+        self.log("test_acc", self.test_accuracy, prog_bar=True)
+
+    def configure_optimizers(self) -> torch.optim.Optimizer:
+        optimizer = torch.optim.Adam(self.parameters(), lr=self.learning_rate)
+        return optimizer
+
+    def train_dataloader(self) -> DataLoader:
+        return self.dataloader_train
+
+    def test_dataloader(self) -> DataLoader:
+        return self.dataloader_test
+
+
+@step(enable_cache=True)
+def trainer(
+    dataloader_train: DataLoader,
+    dataloader_test: DataLoader,
+) -> nn.Module:
+    """Trains on the train dataloader."""
+    pretrained_model, _, _ = load_model()
+
+    # uncomment if you want to re-train model
```

Comment:
This step doesn't really train anything in it's current state, right?

Reply:
Yes, because I wanted to just demonstrate the XAI step, and for that, I need a model. If you uncomment this code you will retrain the model on a different dataset and probably explanations will give more reasonable explanations.

Reply:
Maybe makes sense to add a parameter for it then?

Reply:
I've changed `batch_size` parameter to speed up training and set fine-tuning as a non-optional step in the pipeline: https://github.com/zenml-io/zenml/pull/1663/commits/a044fb5f259e54c53f5950829f8804749dca8ba3.